script.js

```diff
@@ -1 +1,191 @@
+        if (selected) {
+            selected.classList.remove('selected');
+        }
+        dayElement.classList.add('selected');
+    }
+
+    // load To Do List
+    function loadTasks(dateKey) {
+        taskList.innerHTML = ''; // 기존 할 일 목록 지우기
+        const tasks = JSON.parse(localStorage.getItem(dateKey)) || []; // LocalStorage에서 해당 날짜의 할 일 가져오기
+
+        tasks.forEach((task, index) => {
+            addTaskToDOM(task, index);
+        });
+    }
+
+
+    // Add To Do List -> DOM에 추가 -> 화면에 띄우기용
+    function addTaskToDOM(task, index) {
+        const taskElement = document.createElement('article');
+        taskElement.className = 'task';
+        taskElement.innerHTML = `
+            <div class="task-detail">
+                <span class="task-time">${task.time}</span>
+                <h3 class="task-title">${task.title}</h3>
+                <p class="task-desc">${task.desc}</p>
+                <button class="task-delete" data-index="${index}">delete</button>
+            </div>
+        `;
+        taskElement.querySelector('.task-delete').addEventListener('click', function () {
+            const confirm = window.confirm('삭제하시겠습니까?');
+            if (confirm) {
+                deleteTask(selectedDate, index); // 삭제 버튼 클릭 시 할 일 삭제
+            }
+        });
+        taskList.appendChild(taskElement);
+    }
+
+    // Add Task -> LocalStorage에 추가
+    function addTaskToLocal(dateKey, task) {
+        console.log(dateKey);
+        const tasks = JSON.parse(localStorage.getItem(dateKey)) || [];
+
+        // 작성시간
+        task.createdTime = new Date().toLocaleString('en-US', { year: 'numeric', month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' });
+
+        tasks.push(task); 
+        localStorage.setItem(dateKey, JSON.stringify(tasks)); 
+    }
+
+    // Delete Task -> LocalStorage에서 삭제 -> 기존 목록 불러오고, 목록에서 삭제 후 삭제된 목록 저장...흠
+    function deleteTask(dateKey, taskIndex) {
+        const tasks = JSON.parse(localStorage.getItem(dateKey)) || []; 
+        tasks.splice(taskIndex, 1); 
+        localStorage.setItem(dateKey, JSON.stringify(tasks)); 
+        loadTasks(dateKey); 
+    }
+
+    // Modal Button
+    modalBtn.addEventListener('click', () => {
+        modalBtn.style.display = 'none';
+        taskAddForm.style.display = 'flex';
+    });
+
+    // back navigation
+    backBtn.addEventListener('click', () => {
+        modalBtn.style.display = 'block';
+        taskAddForm.style.display = 'none';
+    });
+
+    // Save Task -> submit 이벤트가 폼을 자동으로 제출하면서 페이지가 새로고침됨 -> 그렇기 때문에 save button이 아닌
+    // form에 이벤트리스너를 등록해서 기본동작을 막아야 함. 
+    taskAddForm.addEventListener('submit', function (event) {
+        event.preventDefault();
+
+        const taskTitleInput = document.getElementById('task-title');
+        const taskDescInput = document.getElementById('task-desc');
+        const taskTitle = taskTitleInput.value; 
+        const taskDesc = taskDescInput.value;
```

Comment:
```suggestion
        const taskTitle = document.getElementById('task-title').value ;
        const taskDesc = document.getElementById('task-desc').value;
```
위의 방식으로 코드를 더 간결하게 줄일 수도 있을 것 같아요

Reply:
좋은 의견 감사합니다! 다만 알려주신 코드로 진행했을 때 제 코드 밑에 input field 를 ''로 초기화하는 부분이 있는데 const로 taskTitle을 DOM객체가 아니라 value값 즉, 변수를 선언했기 때문에 밑에서 ''로 초기화할 때 const로 선언된 변수를 새로 선언하는 에러가 발생했어서 DOM 객체와 value를 분리시켜 적용했었습니다! 혹시 제 코드 방법이 아니라 다른 방법으로 input field를 초기화할 수 있는 방안이 있다면 심심하실 때 알려주십셔!

Reply:
추가로 가지신 의문점에 대해 저도 생각 하나 남겨봅니다!
위에서 말씀해주신대로 코드를 간결하게 줄인다면 아래쪽 초기화 부분은
```
document.getElementById('task-title').value='';
document.getElementById('task-desc').value = '';
```
와 같이 DOM 객체에 직접 접근하는 식으로 바로 쓸 수 있을 것 같습니다! 다만 이렇게 되면 반복적으로 DOM을 탐색하게 되어 성능 면에서는 기존대로 한 번 DOM 접근을 통해 저장하는 방식이 규모가 커진 경우에는 좀 더 유리하지 않을까 싶습니다. 물론 여기서는 코드의 가독성을 위해 줄여도 좋겠다고 생각합니다! 초기화 부분에 대해 또 다른 방안이 있다면 저도 무척 궁금하네요 !!

Reply:
DOM객체에 직접 접근하는 방식도 좋은 것 같습니다! 다른 방법을 찾아보았는데 이번처럼 title, desc 모든 form의 입력필드를 초기화할 때 `  taskAddForm.reset();` 이 코드처럼 form 내의 모든 요소를 초기화하는 방법도 있는 것 같습니다. 다만 말 그대로 모든 요소를 초기화하는 것이기 때문에 이번처럼 조건이 맞을 때만 사용해야 할 것 같아요!